examples/webxr_vr_layers.html

```diff
@@ -0,0 +1,171 @@
+				renderer.xr.enabled = true;
+
+				container.appendChild( renderer.domElement );
+
+				document.body.appendChild( VRButton.createButton( renderer ) );
+
+				// controllers
+
+				controller1 = renderer.xr.getController( 0 );
+				scene.add( controller1 );
+
+				controller2 = renderer.xr.getController( 1 );
+				scene.add( controller2 );
+
+				const controllerModelFactory = new XRControllerModelFactory();
+				const handModelFactory = new XRHandModelFactory().setPath( "./models/fbx/" );
+
+				// Hand 1
+				controllerGrip1 = renderer.xr.getControllerGrip( 0 );
+				controllerGrip1.add( controllerModelFactory.createControllerModel( controllerGrip1 ) );
+				scene.add( controllerGrip1 );
+
+				hand1 = renderer.xr.getHand( 0 );
+				hand1.add( handModelFactory.createHandModel( hand1 ) );
+
+				scene.add( hand1 );
+
+				// Hand 2
+				controllerGrip2 = renderer.xr.getControllerGrip( 1 );
+				controllerGrip2.add( controllerModelFactory.createControllerModel( controllerGrip2 ) );
+				scene.add( controllerGrip2 );
+
+				hand2 = renderer.xr.getHand( 1 );
+				hand2.add( handModelFactory.createHandModel( hand2 ) );
+				scene.add( hand2 );
+
+				//
+
+				const geometry = new THREE.BufferGeometry().setFromPoints( [ new THREE.Vector3( 0, 0, 0 ), new THREE.Vector3( 0, 0, - 1 ) ] );
+
+				const line = new THREE.Line( geometry );
+				line.name = 'line';
+				line.scale.z = 5;
+
+				controller1.add( line.clone() );
+				controller2.add( line.clone() );
+
+				//
+
+				window.addEventListener( 'resize', onWindowResize, false );
+
+				video = document.createElement('video');
+				video.loop = true;
+				video.src = 'textures/pano.webm';
+
+				video2 = document.createElement('video');
+				video2.loop = true;
+				video2.src = 'textures/MaryOculus.webm';
+			}
+
+			function onWindowResize() {
+
+				camera.aspect = window.innerWidth / window.innerHeight;
+				camera.updateProjectionMatrix();
+
+				renderer.setSize( window.innerWidth, window.innerHeight );
+
+			}
+			//
+
+			function animate() {
+
+				renderer.setAnimationLoop( render );
+
+			}
+
+			function render() {
+				const xr = renderer.xr;
+				const session = xr.getSession();
+
+				if ( session && session.renderState.layers !== undefined && session.hasMediaLayer === undefined && video.readyState >= 2 && video2.readyState >= 2) {
+					session.hasMediaLayer = true;
+					session.requestReferenceSpace('local').then((refSpace) => {
+						const mediaBinding = new XRMediaBinding(session);
+						const equirectLayer = mediaBinding.createEquirectLayer(video, {space: refSpace, layout: "mono"});
+						const quadLayer = mediaBinding.createQuadLayer(video2, {space: refSpace, layout: "stereo-left-right"});
+						quadLayer.transform = new XRRigidTransform({x: 1.5, y: 1.0, z: -2.0});
+						session.updateRenderState( { layers: [ equirectLayer, quadLayer, session.renderState.layers[0] ] } );
+						video.play();
+						video2.play();
+					});
+				}
```

Comment:
I think we could create a `XRMediaLayer` class that extends `Object3D` so we can add it to the scene like any other object. And then let `WebXRManager` do all this binging stuff when it sees a `XRMediaLayer`.

What do you think of that design?

Reply:
That sounds reasonable.
How would you handle WebGL Layers?

Reply:
> How would you handle WebGL Layers?

Can you elaborate?

Reply:
WebGL layers give you another texture to render to.
Just like how a projection layer gives you a stereo texture, other layer types can give you one.
Can three.js draw into multiple textures at a time? non-projection layers are also special in that they don't have to be drawn at the refresh rate.

Reply:
No, it can't draw into multiple textures yet.

I was not aware of WebGL layers... Seems like I'll have to do a proper read at the spec. I'm too confused right now 😅

Reply:
We can tackle those later :-)
I believe this project updated three.js so it could target multiple layers at the same time: https://spellew.github.io/webxr-samples/media-layers.html

Reply:
Is there a talk that explains Layers? Or is https://immersive-web.github.io/layers/ the only resource?

Reply:
That is the only resource for the WebXR part.
For general information on layers, you can read:
- https://developer.oculus.com/documentation/unity/unity-ovroverlay/?locale=en_US
- https://developer.oculus.com/documentation/unreal/unreal-overlay/?locale=en_US
- https://www.khronos.org/registry/OpenXR/specs/1.0/html/xrspec.html#rendering

Reply:
@dmarcos also hooked them into aframe: https://twitter.com/dmarcos/status/1322309762562760704